### PR TITLE
fix(router): state data missing in routerLink

### LINKED ISSUE #33173: state input of RouterLink does not work
# 🐞 bug report

### Affected Package
@angular/router: 8.1.2
@angular/router: 8.2.10

### Is this a regression?

No

### Description

state input of RouterLink does not work. However, Router.navigate() with state specified in NavigationExtras does work.

## 🔬 Minimal Reproduction
https://stackblitz.com/edit/angular-szomyj

## 🌍  Your Environment

**Angular Version:**
<pre><code>

Angular CLI: 8.1.2
Node: 10.16.0
OS: linux x64
Angular: 8.1.2
... animations, cli, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                            Version
------------------------------------------------------------
@angular-devkit/architect          0.801.2
@angular-devkit/build-angular      0.801.2
@angular-devkit/build-ng-packagr   0.801.2
@angular-devkit/build-optimizer    0.801.2
@angular-devkit/build-webpack      0.801.2
@angular-devkit/core               8.1.2
@angular-devkit/schematics         8.1.2
@angular/cdk                       8.1.1
@angular/material                  8.1.1
@angular/material-moment-adapter   8.1.1
@ngtools/webpack                   8.1.2
@schematics/angular                8.1.2
@schematics/update                 0.801.2
ng-packagr                         5.4.0
rxjs                               6.5.2
typescript                         3.4.5
webpack                            4.35.2

</code></pre>

**Anything else relevant?**
Also doesn't work in "@angular/router": "8.2.10", see the StackBlitz

### PR DESCRIPTION
**Pull request by atscott**
fixes #33173 - router state data is missing on routerLink when used
with non-anchor elements.

### CODE REVIEW

File: packages/router/test/integration.spec.ts [hunk 1]
```diff
@@ -2019,35 +2019,43 @@ describe('Integration', () => {
...
+      });
 
-         router.resetConfig([{
-           path: 'team/:id',
-           component: TeamCmp,
-           children: [
-             {path: 'link', component: LinkWithState}, {path: 'simple', component: SimpleCmp}
-           ]
-         }]);
+      it('for non-anchor elements', ()=> {
```
Comment (AndrewKushnir) [inline packages/router/test/integration.spec.ts:2028]: Missing space after `()` (lint will probably catch this too).

File: packages/router/test/integration.spec.ts [hunk 2]
```diff
@@ -2019,35 +2019,43 @@ describe('Integration', () => {
          expect(location.path()).toEqual('/team/22/simple?q=1#f');
        })));
 
-    it('should support history state',
-       fakeAsync(inject([Router, Location], (router: Router, location: SpyLocation) => {
-         const fixture = createRoot(router, RootCmp);
+    describe('should support history and state', () => {
+      let component: typeof LinkWithState|typeof DivLinkWithState;
+      it('for anchor elements', () => {
+        component = LinkWithState;
```
Comment (AndrewKushnir) [inline packages/router/test/integration.spec.ts:2025]: That's an interesting approach!

I was a bit confused when I first looked at the change. May be add a quick comment saying that the actual validation happens in `afterEach` hook?

**Review by AndrewKushnir — APPROVED** [review]
LGTM with a couple nits 👍 Thanks @atscott.

**Comment by atscott** [comment]
[presubmit](https://test.corp.google.com/ui#id=OCL:305119267:BASE:305119313:1586208873672:412b3882)

**Review by atscott — COMMENTED** [review]
merge assistance: I'm the owner of router and can't approve my own PR. Review from Andrew K should be enough.

**Pull request closed without merging**